{"code":"ECONNREFUSED","fatal":true,"level":"error","message":"Database connection failed","stack":"AggregateError [ECONNREFUSED]: \n    at internalConnectMultiple (node:net:1118:18)\n    at afterConnectMultiple (node:net:1685:7)","timestamp":"2025-03-12T13:34:31.765Z"}
{"code":"ECONNREFUSED","fatal":true,"level":"error","message":"Failed to connect to database","stack":"AggregateError [ECONNREFUSED]: \n    at internalConnectMultiple (node:net:1118:18)\n    at afterConnectMultiple (node:net:1685:7)","timestamp":"2025-03-12T13:34:31.770Z"}
{"code":"ECONNREFUSED","fatal":true,"level":"error","message":"Database connection failed","stack":"AggregateError [ECONNREFUSED]: \n    at internalConnectMultiple (node:net:1118:18)\n    at afterConnectMultiple (node:net:1685:7)","timestamp":"2025-03-12T13:38:09.896Z"}
{"code":"ECONNREFUSED","fatal":true,"level":"error","message":"Failed to connect to database","stack":"AggregateError [ECONNREFUSED]: \n    at internalConnectMultiple (node:net:1118:18)\n    at afterConnectMultiple (node:net:1685:7)","timestamp":"2025-03-12T13:38:09.976Z"}
{"level":"error","message":"Error registering User data and salt arguments required","stack":"Error: data and salt arguments required\n    at Object.hash (D:\\expensa\\backend\\node_modules\\bcrypt\\bcrypt.js:137:17)\n    at D:\\expensa\\backend\\node_modules\\bcrypt\\promises.js:29:12\n    at new Promise (<anonymous>)\n    at module.exports.promise (D:\\expensa\\backend\\node_modules\\bcrypt\\promises.js:20:12)\n    at Object.hash (D:\\expensa\\backend\\node_modules\\bcrypt\\bcrypt.js:133:25)\n    at registerUser (D:\\expensa\\backend\\controllers\\userController.js:23:45)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:23:45.092Z"}
{"level":"error","message":"Error: data and salt arguments required\n    at Object.hash (D:\\expensa\\backend\\node_modules\\bcrypt\\bcrypt.js:137:17)\n    at D:\\expensa\\backend\\node_modules\\bcrypt\\promises.js:29:12\n    at new Promise (<anonymous>)\n    at module.exports.promise (D:\\expensa\\backend\\node_modules\\bcrypt\\promises.js:20:12)\n    at Object.hash (D:\\expensa\\backend\\node_modules\\bcrypt\\bcrypt.js:133:25)\n    at registerUser (D:\\expensa\\backend\\controllers\\userController.js:23:45)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:23:45.094Z"}
{"level":"error","message":"Error logging in User secretOrPrivateKey must have a value","stack":"Error: secretOrPrivateKey must have a value\n    at module.exports [as sign] (D:\\expensa\\backend\\node_modules\\jsonwebtoken\\sign.js:111:20)\n    at loginUser (D:\\expensa\\backend\\controllers\\userController.js:57:29)","timestamp":"2025-03-25T10:34:17.134Z"}
{"level":"error","message":"Error: secretOrPrivateKey must have a value\n    at module.exports [as sign] (D:\\expensa\\backend\\node_modules\\jsonwebtoken\\sign.js:111:20)\n    at loginUser (D:\\expensa\\backend\\controllers\\userController.js:57:29)","timestamp":"2025-03-25T10:34:17.135Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = '1856', reset_token_expiry = 1742899183069 WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":["1856",1742899183069,null],"sql":"UPDATE user_auth SET reset_token = ?, reset_token_expiry = ? WHERE id = ?","timestamp":"2025-03-25T10:39:42.772Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error requesting password reset Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = '1856', reset_token_expiry = 1742899183069 WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:39:42.774Z"}
{"level":"error","message":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:39:42.776Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = '7906', reset_token_expiry = 1742899242344 WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":["7906",1742899242344,null],"sql":"UPDATE user_auth SET reset_token = ?, reset_token_expiry = ? WHERE id = ?","timestamp":"2025-03-25T10:40:42.045Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error requesting password reset Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = '7906', reset_token_expiry = 1742899242344 WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:40:42.046Z"}
{"level":"error","message":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:40:42.048Z"}
{"error":{"code":"ER_TRUNCATED_WRONG_VALUE","errno":1292,"message":"Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1","sql":"UPDATE user_auth SET reset_token = '1636', reset_token_expiry = 1742899303564 WHERE uid = 1","sqlMessage":"Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1","sqlState":"22007"},"level":"error","message":"Error executing SQL query","params":["1636",1742899303564,1],"sql":"UPDATE user_auth SET reset_token = ?, reset_token_expiry = ? WHERE uid = ?","timestamp":"2025-03-25T10:41:43.267Z"}
{"code":"ER_TRUNCATED_WRONG_VALUE","errno":1292,"level":"error","message":"Error requesting password reset Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1","sql":"UPDATE user_auth SET reset_token = '1636', reset_token_expiry = 1742899303564 WHERE uid = 1","sqlMessage":"Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1","sqlState":"22007","stack":"Error: Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:41:43.269Z"}
{"level":"error","message":"Error: Incorrect time value: '1742899303564' for column 'reset_token_expiry' at row 1\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.setOtp (D:\\expensa\\backend\\models\\user.js:32:12)\n    at requestPasswordReset (D:\\expensa\\backend\\controllers\\userController.js:82:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T10:41:43.271Z"}
{"level":"error","message":"Error resetting password User.getByOtp is not a function","stack":"TypeError: User.getByOtp is not a function\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:47:12)","timestamp":"2025-03-25T10:47:14.350Z"}
{"level":"error","message":"TypeError: User.getByOtp is not a function\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:280:10)\n    at Function.handle (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:175:3)\n    at router (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:47:12)","timestamp":"2025-03-25T10:47:14.354Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'reset_otp' in 'where clause'","sql":"SELECT * FROM user_auth WHERE reset_otp = 5127 AND reset_token_expiry > 1742900242813","sqlMessage":"Unknown column 'reset_otp' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":[5127,1742900242813],"sql":"SELECT * FROM user_auth WHERE reset_otp = ? AND reset_token_expiry > ?","timestamp":"2025-03-25T10:57:22.823Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error resetting password Unknown column 'reset_otp' in 'where clause'","sql":"SELECT * FROM user_auth WHERE reset_otp = 5127 AND reset_token_expiry > 1742900242813","sqlMessage":"Unknown column 'reset_otp' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'reset_otp' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.getByOtp (D:\\expensa\\backend\\models\\user.js:53:27)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T10:57:22.825Z"}
{"level":"error","message":"Error: Unknown column 'reset_otp' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.getByOtp (D:\\expensa\\backend\\models\\user.js:53:27)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T10:57:22.826Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'reset_otp' in 'where clause'","sql":"SELECT * FROM user_auth WHERE reset_otp = 8500 AND reset_token_expiry > 1742900317436","sqlMessage":"Unknown column 'reset_otp' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":[8500,1742900317436],"sql":"SELECT * FROM user_auth WHERE reset_otp = ? AND reset_token_expiry > ?","timestamp":"2025-03-25T10:58:37.437Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error resetting password Unknown column 'reset_otp' in 'where clause'","sql":"SELECT * FROM user_auth WHERE reset_otp = 8500 AND reset_token_expiry > 1742900317436","sqlMessage":"Unknown column 'reset_otp' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'reset_otp' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.getByOtp (D:\\expensa\\backend\\models\\user.js:53:27)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T10:58:37.439Z"}
{"level":"error","message":"Error: Unknown column 'reset_otp' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.getByOtp (D:\\expensa\\backend\\models\\user.js:53:27)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:107:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T10:58:37.442Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'user_id' in 'where clause'","sql":"UPDATE user_auth set password = '$2b$10$C2r06IY/1altBpTMAtN.LuDCrhzEpNuUZZDFZ1pTQB4Aq3qSVK8qy' where user_id = NULL","sqlMessage":"Unknown column 'user_id' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":["$2b$10$C2r06IY/1altBpTMAtN.LuDCrhzEpNuUZZDFZ1pTQB4Aq3qSVK8qy",null],"sql":"UPDATE user_auth set password = ? where user_id = ?","timestamp":"2025-03-25T11:00:38.948Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error resetting password Unknown column 'user_id' in 'where clause'","sql":"UPDATE user_auth set password = '$2b$10$C2r06IY/1altBpTMAtN.LuDCrhzEpNuUZZDFZ1pTQB4Aq3qSVK8qy' where user_id = NULL","sqlMessage":"Unknown column 'user_id' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'user_id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updatePassword (D:\\expensa\\backend\\models\\user.js:47:12)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:125:20)","timestamp":"2025-03-25T11:00:38.949Z"}
{"level":"error","message":"Error: Unknown column 'user_id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updatePassword (D:\\expensa\\backend\\models\\user.js:47:12)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:125:20)","timestamp":"2025-03-25T11:00:38.950Z"}
{"level":"error","message":"Error resetting password User.clearOtp is not a function","stack":"TypeError: User.clearOtp is not a function\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:126:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T11:01:03.445Z"}
{"level":"error","message":"TypeError: User.clearOtp is not a function\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:126:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T11:01:03.446Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = NULL, reset_token_expiry = NULL WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":[null],"sql":"UPDATE user_auth SET reset_token = NULL, reset_token_expiry = NULL WHERE id = ?","timestamp":"2025-03-25T11:01:40.463Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error resetting password Unknown column 'id' in 'where clause'","sql":"UPDATE user_auth SET reset_token = NULL, reset_token_expiry = NULL WHERE id = NULL","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.clearResetToken (D:\\expensa\\backend\\models\\user.js:39:12)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:126:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T11:01:40.465Z"}
{"level":"error","message":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.clearResetToken (D:\\expensa\\backend\\models\\user.js:39:12)\n    at resetPassword (D:\\expensa\\backend\\controllers\\userController.js:126:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-25T11:01:40.466Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-25T12:01:40.015Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000","stack":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:01:40.017Z"}
{"level":"error","message":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:01:40.019Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'country' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'country' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-25T12:13:58.540Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'country' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'country' cannot be null","sqlState":"23000","stack":"Error: Column 'country' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:13:58.542Z"}
{"level":"error","message":"Error: Column 'country' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:13:58.544Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-25T12:19:58.189Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000","stack":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:19:58.192Z"}
{"level":"error","message":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:33:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:19:58.197Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-25T12:20:49.795Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000","stack":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:20:49.797Z"}
{"level":"error","message":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-25T12:20:49.799Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-26T03:54:20.552Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000","stack":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-26T03:54:20.554Z"}
{"level":"error","message":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-26T03:54:20.558Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":[null,null,null,null,500,null,null],"sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (?, ?, ?, ?, ?, ?, ?)","timestamp":"2025-03-26T04:00:12.578Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating profile Column 'username' cannot be null","sql":"INSERT INTO user_profile (username, phone_number, dob, profile_img, xpensa_coins, country, user_id) VALUES (NULL, NULL, NULL, NULL, 500, NULL, NULL)","sqlMessage":"Column 'username' cannot be null","sqlState":"23000","stack":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-26T04:00:12.580Z"}
{"level":"error","message":"Error: Column 'username' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createProfile (D:\\expensa\\backend\\models\\profile.js:5:29)\n    at addProfile (D:\\expensa\\backend\\controllers\\profileController.js:34:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:346:12)","timestamp":"2025-03-26T04:00:12.583Z"}
{"level":"error","message":"MulterError: Unexpected field\n    at wrappedFileFilter (D:\\expensa\\backend\\node_modules\\multer\\index.js:40:19)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:109:7)\n    at Multipart.emit (node:events:519:28)\n    at HeaderParser.cb (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:358:14)\n    at HeaderParser.push (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:162:20)\n    at SBMH.ssCb [as _cb] (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:394:37)\n    at feed (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:248:10)\n    at SBMH.push (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:104:16)\n    at Multipart._write (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:567:19)\n    at writeOrBuffer (node:internal/streams/writable:570:12)","timestamp":"2025-03-26T04:15:26.211Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\images\\http:\\localhost:undefined\\profile-image\\20250326T041545269Z_hero.jpg'\n    at Object.openSync (node:fs:573:18)\n    at Object.writeFileSync (node:fs:2381:35)\n    at saveProfileImages (D:\\expensa\\backend\\utils\\saveProfileImages.js:21:8)\n    at D:\\expensa\\backend\\routes\\profileRoutes.js:22:32\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)","timestamp":"2025-03-26T04:15:45.276Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\images\\http:\\localhost:undefined\\images\\20250326T041633383Z_hero.jpg'\n    at Object.openSync (node:fs:573:18)\n    at Object.writeFileSync (node:fs:2381:35)\n    at saveProfileImages (D:\\expensa\\backend\\utils\\saveProfileImages.js:21:8)\n    at D:\\expensa\\backend\\routes\\profileRoutes.js:22:32\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)","timestamp":"2025-03-26T04:16:33.388Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\images\\http:\\localhost:3000\\images\\20250326T041801305Z_hero.jpg'\n    at Object.openSync (node:fs:573:18)\n    at Object.writeFileSync (node:fs:2381:35)\n    at saveProfileImages (D:\\expensa\\backend\\utils\\saveProfileImages.js:21:8)\n    at D:\\expensa\\backend\\routes\\profileRoutes.js:22:32\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)","timestamp":"2025-03-26T04:18:01.310Z"}
{"code":"ERR_INVALID_ARG_VALUE","level":"error","message":"Error during bill extraction: The argument 'path' must be a string, Uint8Array, or URL without null bytes. Received <Buffer ff d8 ff e0 00 10 4a 46 49 46 00 01 01 01 00 60 00 60 00 00 ff db 00 43 00 03 02 02 03 02 02 03 03 03 03 04 03 03 04 05 ...","stack":"TypeError [ERR_INVALID_ARG_VALUE]: The argument 'path' must be a string, Uint8Array, or URL without null bytes. Received <Buffer ff d8 ff e0 00 10 4a 46 49 46 00 01 01 01 00 60 00 60 00 00 ff db 00 43 00 03 02 02 03 02 02 03 03 03 03 04 03 03 04 05 ...\n    at Object.openSync (node:fs:571:10)\n    at Object.readFileSync (node:fs:452:35)\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:18:28)\n    at getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:70:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)","timestamp":"2025-03-26T11:13:03.259Z"}
{"level":"error","message":"Error extracting expense details Failed to extract bill details","stack":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:60:11)\n    at getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:70:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)\n    at emitCloseNT (node:internal/streams/destroy:147:10)\n    at process.processTicksAndRejections (node:internal/process/task_queues:81:21)","timestamp":"2025-03-26T11:13:03.264Z"}
{"level":"error","message":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:60:11)\n    at getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:70:33)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at done (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:47:7)\n    at indicateDone (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:51:68)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:168:7)\n    at Multipart.emit (node:events:519:28)\n    at emitCloseNT (node:internal/streams/destroy:147:10)\n    at process.processTicksAndRejections (node:internal/process/task_queues:81:21)","timestamp":"2025-03-26T11:13:03.270Z"}
{"level":"error","message":"Error extracting expense details Failed to extract bill details","stack":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:62:11)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:73:27)","timestamp":"2025-03-26T11:14:43.399Z"}
{"level":"error","message":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:62:11)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:73:27)","timestamp":"2025-03-26T11:14:43.406Z"}
{"level":"error","message":"Error extracting expense details Failed to extract bill details","stack":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:64:11)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:75:27)","timestamp":"2025-03-26T11:15:32.379Z"}
{"level":"error","message":"Error: Failed to extract bill details\n    at extractBillDetails (D:\\expensa\\backend\\controllers\\expenseController.js:64:11)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async getExpenseDetails (D:\\expensa\\backend\\controllers\\expenseController.js:75:27)","timestamp":"2025-03-26T11:15:32.391Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\billImage\\20250326T113008428Z_test2.jpg'","timestamp":"2025-03-26T11:30:08.442Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\billImage\\20250326T113014500Z_test2.jpg'","timestamp":"2025-03-26T11:30:14.504Z"}
{"level":"error","message":"MulterError: Unexpected field\n    at wrappedFileFilter (D:\\expensa\\backend\\node_modules\\multer\\index.js:40:19)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:109:7)\n    at Multipart.emit (node:events:519:28)\n    at HeaderParser.cb (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:358:14)\n    at HeaderParser.push (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:162:20)\n    at SBMH.ssCb [as _cb] (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:394:37)\n    at feed (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:248:10)\n    at SBMH.push (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:104:16)\n    at Multipart._write (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:567:19)\n    at writeOrBuffer (node:internal/streams/writable:570:12)","timestamp":"2025-03-26T11:37:56.606Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\billImage\\20250326T113926640Z_test1.jpg'","timestamp":"2025-03-26T11:39:26.645Z"}
{"level":"error","message":"Error: ENOENT: no such file or directory, open 'D:\\expensa\\backend\\expensa\backend\billsImage\\20250326T114150084Z_test1.jpg'","timestamp":"2025-03-26T11:41:50.088Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'category' in 'field list'","sql":"\n    INSERT INTO expense_record (user_id, category, amount, date, description, bill_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":["1",null,"1050","2024-06-15","this is testing",null],"sql":"\n    INSERT INTO expense_record (user_id, category, amount, date, description, bill_image)\n    VALUES (?, ?, ?, ?, ?, ?)\n  ","timestamp":"2025-03-26T11:42:11.313Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error creating expense Unknown column 'category' in 'field list'","sql":"\n    INSERT INTO expense_record (user_id, category, amount, date, description, bill_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22","stack":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:42:11.315Z"}
{"level":"error","message":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:42:11.318Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'category_id' cannot be null","sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Column 'category_id' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":["1",null,"1050","2024-06-15","this is testing",null],"sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES (?, ?, ?, ?, ?, ?)\n  ","timestamp":"2025-03-26T11:52:51.276Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating expense Column 'category_id' cannot be null","sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Column 'category_id' cannot be null","sqlState":"23000","stack":"Error: Column 'category_id' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:52:51.281Z"}
{"level":"error","message":"Error: Column 'category_id' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:52:51.286Z"}
{"error":{"code":"ER_BAD_NULL_ERROR","errno":1048,"message":"Column 'category_id' cannot be null","sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Column 'category_id' cannot be null","sqlState":"23000"},"level":"error","message":"Error executing SQL query","params":["1",null,"1050","2024-06-15","this is testing",null],"sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES (?, ?, ?, ?, ?, ?)\n  ","timestamp":"2025-03-26T11:54:36.316Z"}
{"code":"ER_BAD_NULL_ERROR","errno":1048,"level":"error","message":"Error creating expense Column 'category_id' cannot be null","sql":"\n    INSERT INTO expense_record (user_id, category_id, amount, date, description, receipt_image)\n    VALUES ('1', NULL, '1050', '2024-06-15', 'this is testing', NULL)\n  ","sqlMessage":"Column 'category_id' cannot be null","sqlState":"23000","stack":"Error: Column 'category_id' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:54:36.319Z"}
{"level":"error","message":"Error: Column 'category_id' cannot be null\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.createExpense (D:\\expensa\\backend\\models\\expenseModel.js:11:29)\n    at makeExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:90:43)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:54:36.323Z"}
{"level":"error","message":"Error updating expense user_id is not defined","stack":"ReferenceError: user_id is not defined\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:146:53)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:59:28.493Z"}
{"level":"error","message":"ReferenceError: user_id is not defined\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:146:53)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T11:59:28.497Z"}
{"level":"error","message":"Error updating expense user_id is not defined","stack":"ReferenceError: user_id is not defined\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:146:53)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:01:41.993Z"}
{"level":"error","message":"ReferenceError: user_id is not defined\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:146:53)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:01:41.996Z"}
{"level":"error","message":"Error updating expense Assignment to constant variable.","stack":"TypeError: Assignment to constant variable.\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:147:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:01:51.315Z"}
{"level":"error","message":"TypeError: Assignment to constant variable.\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:147:20)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:01:51.316Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'category' in 'field list'","sql":"\n    UPDATE expense_record\n    SET category = NULL,\n        amount = 250.75,\n        date = '2024-03-26',\n        description = 'Dinner at a restaurant',\n        bill_image = ''\n    WHERE eid = 2\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":[null,250.75,"2024-03-26","Dinner at a restaurant","",2],"sql":"\n    UPDATE expense_record\n    SET category = ?,\n        amount = ?,\n        date = ?,\n        description = ?,\n        bill_image = ?\n    WHERE eid = ?\n  ","timestamp":"2025-03-26T12:02:09.931Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error updating expense Unknown column 'category' in 'field list'","sql":"\n    UPDATE expense_record\n    SET category = NULL,\n        amount = 250.75,\n        date = '2024-03-26',\n        description = 'Dinner at a restaurant',\n        bill_image = ''\n    WHERE eid = 2\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22","stack":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:35:12)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:156:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:02:09.933Z"}
{"level":"error","message":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:35:12)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:156:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:02:09.938Z"}
{"level":"error","message":"Error updating expense bill_image is not defined","stack":"ReferenceError: bill_image is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:39:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:03:09.711Z"}
{"level":"error","message":"ReferenceError: bill_image is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:39:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T12:03:09.714Z"}
{"level":"error","message":"Error updating expense bill_image is not defined","stack":"ReferenceError: bill_image is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:39:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:03:29.997Z"}
{"level":"error","message":"ReferenceError: bill_image is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:39:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:03:29.999Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'category' in 'field list'","sql":"\n    UPDATE expense_record\n    SET category = NULL,\n        amount = 250.75,\n        date = '2024-03-26',\n        description = 'Dinner at a restaurant'\n    WHERE eid = 2\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":[null,250.75,"2024-03-26","Dinner at a restaurant",2],"sql":"\n    UPDATE expense_record\n    SET category = ?,\n        amount = ?,\n        date = ?,\n        description = ?\n    WHERE eid = ?\n  ","timestamp":"2025-03-26T14:04:16.811Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error updating expense Unknown column 'category' in 'field list'","sql":"\n    UPDATE expense_record\n    SET category = NULL,\n        amount = 250.75,\n        date = '2024-03-26',\n        description = 'Dinner at a restaurant'\n    WHERE eid = 2\n  ","sqlMessage":"Unknown column 'category' in 'field list'","sqlState":"42S22","stack":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:34:12)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:04:16.813Z"}
{"level":"error","message":"Error: Unknown column 'category' in 'field list'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:34:12)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:04:16.816Z"}
{"level":"error","message":"Error updating expense category is not defined","stack":"ReferenceError: category is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:35:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:04:46.098Z"}
{"level":"error","message":"ReferenceError: category is not defined\n    at Object.updateExpense (D:\\expensa\\backend\\models\\expenseModel.js:35:5)\n    at editExpenseEntry (D:\\expensa\\backend\\controllers\\expenseController.js:151:47)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","timestamp":"2025-03-26T14:04:46.101Z"}
{"level":"error","message":"Error fetching category details categoryId is not defined","stack":"ReferenceError: categoryId is not defined\n    at getCategoryDetails (D:\\expensa\\backend\\controllers\\categoryController.js:9:9)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:280:10)","timestamp":"2025-03-26T14:37:36.938Z"}
{"level":"error","message":"ReferenceError: categoryId is not defined\n    at getCategoryDetails (D:\\expensa\\backend\\controllers\\categoryController.js:9:9)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:365:14)\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:376:14)\n    at Function.process_params (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:421:3)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:280:10)","timestamp":"2025-03-26T14:37:36.941Z"}
{"level":"error","message":"MulterError: Unexpected field\n    at wrappedFileFilter (D:\\expensa\\backend\\node_modules\\multer\\index.js:40:19)\n    at Multipart.<anonymous> (D:\\expensa\\backend\\node_modules\\multer\\lib\\make-middleware.js:109:7)\n    at Multipart.emit (node:events:519:28)\n    at HeaderParser.cb (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:358:14)\n    at HeaderParser.push (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:162:20)\n    at SBMH.ssCb [as _cb] (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:394:37)\n    at feed (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:200:10)\n    at SBMH.push (D:\\expensa\\backend\\node_modules\\streamsearch\\lib\\sbmh.js:104:16)\n    at Multipart._write (D:\\expensa\\backend\\node_modules\\busboy\\lib\\types\\multipart.js:567:19)\n    at writeOrBuffer (node:internal/streams/writable:570:12)","timestamp":"2025-03-26T14:52:06.615Z"}
{"error":{"code":"ER_BAD_FIELD_ERROR","errno":1054,"message":"Unknown column 'id' in 'where clause'","sql":"DELETE FROM expense_record WHERE id = '2'","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22"},"level":"error","message":"Error executing SQL query","params":["2"],"sql":"DELETE FROM expense_record WHERE id = ?","timestamp":"2025-03-26T16:49:15.168Z"}
{"code":"ER_BAD_FIELD_ERROR","errno":1054,"level":"error","message":"Error deleting expense by ID Unknown column 'id' in 'where clause'","sql":"DELETE FROM expense_record WHERE id = '2'","sqlMessage":"Unknown column 'id' in 'where clause'","sqlState":"42S22","stack":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.removeExpenseById (D:\\expensa\\backend\\models\\expenseModel.js:93:29)\n    at deleteExpenseById (D:\\expensa\\backend\\controllers\\expenseController.js:220:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:365:14)","timestamp":"2025-03-26T16:49:15.179Z"}
{"level":"error","message":"Error: Unknown column 'id' in 'where clause'\n    at PromisePool.query (D:\\expensa\\backend\\node_modules\\mysql2\\lib\\promise\\pool.js:36:22)\n    at Object.query (D:\\expensa\\backend\\utils\\db.js:28:39)\n    at Object.removeExpenseById (D:\\expensa\\backend\\models\\expenseModel.js:93:29)\n    at deleteExpenseById (D:\\expensa\\backend\\controllers\\expenseController.js:220:42)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at next (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:149:13)\n    at Route.dispatch (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\route.js:119:3)\n    at Layer.handle [as handle_request] (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\layer.js:95:5)\n    at D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:284:15\n    at param (D:\\expensa\\backend\\node_modules\\express\\lib\\router\\index.js:365:14)","timestamp":"2025-03-26T16:49:15.191Z"}
